### PR TITLE
Generalize Course Schedule scraping 

### PR DESCRIPTION
**Pull request by keithradford**
Got rid of the scraping method which put information into a list and moved to a more regex based solution to generalize this function more.

### CODE REVIEW

**Comment by keithradford** [comment]
Passes all the tests now. Didn't fix the test issue on Windows yet. That's next, but should be ready to merge now.

File: src/pages/courseListingEntries/index.ts [hunk 1]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
 
       // Parse schedule table
       // This must be done before sectionInfo because sectionInfo removes this table from the cheerio object
-      // Having this table in the cheerio object was causing inconsistent errors whilst parsing the prior section, thus I removed it.
+      // Having this table in the cheerio object was causing inconsistent errors whilst parsing the prior section, thus I removed it.\
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:44]: I don't having this comment helps for development in the future. It's more of a comment inside a commit. I'd remove it.

File: src/pages/courseListingEntries/index.ts [hunk 2]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
+          const associatedStart = dayjs(associatedStartRegex.exec(associatedTerm)![1], 'MMM').format('MM');
+          const associatedEnd = dayjs(associatedEndRegex.exec(associatedTerm)![1], 'MMM').format('MM');
+          const year = yearRegex.exec(associatedTerm)![0];
+          section.associatedTerm = {
+            start: year + associatedStart,
+            end: year + associatedEnd,
+          };
+        }
+      }
+      console.log(section.associatedTerm);
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:103]: I don't think we need this.

File: src/pages/courseListingEntries/index.ts [hunk 3]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
-      const associatedStart = dayjs(associatedStartRegex.exec(sectionInfo[1])![1], 'MMM').format('MM');
-      const associatedEnd = dayjs(associatedEndRegex.exec(sectionInfo[1],)![1], 'MMM').format('MM');
-      const year = yearRegex.exec(sectionInfo[1])![0];
-      section.associatedTerm = {
-        start: year + associatedStart,
-        end: year + associatedEnd
-      };
+
+      if (additionalInfoRegex.test(sectionInfo)) {
+        section.additionalInfo = additionalInfoRegex.exec(sectionInfo)![1].trim();
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:82]: A small nit-pick. Shouldn't the regex capture so we don't need to run `trim()`?

File: src/pages/courseListingEntries/index.ts [hunk 4]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
+        if (registrationStartRegex.test(registrationDates) && registrationEndRegex.test(registrationDates)) {
+          const registrationStart = registrationStartRegex.exec(registrationDates)![1].trim();
+          const registrationEnd = registrationEndRegex.exec(registrationDates)![1].trim();
+          section.registrationDates = {
+            start: registrationStart,
+            end: registrationEnd,
+          };
+        }
+      }
+      console.log(section.registrationDates);
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:118]: Same thing here.

File: src/pages/courseListingEntries/index.ts [hunk 5]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
-      const levels = levelsRegex.exec(sectionInfo[3])![1].toLowerCase();
-      section.levels = levels.split(/,\s*/) as levelType[];
+      if (levelsRegex.test(sectionInfo)) {
+        const levels = levelsRegex
+          .exec(sectionInfo)![1]
+          .toLowerCase()
+          .trim();
+        section.levels = levels.split(/,\s*/) as levelType[];
+      }
+      console.log(section.levels);
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:129]: ditto

File: src/pages/courseListingEntries/index.ts [hunk 6]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
 
       // Check if online campus or in-person campus
       // Might change this because in the HTML it's either: online or main campus (might be other campuses too)
-      if(/online/i.test(sectionInfo[4]) === true){
+      // Problem may arise if classes are offered online & in-person, will look into
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:133]: Let's prefix a `TODO` for these kinda things.

File: src/pages/courseListingEntries/index.ts [hunk 7]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
+      if (/online/i.test(sectionInfo)) {
         section.campus = 'online';
-      }
-      else{
+      } else {
         section.campus = 'in-person';
       }
+      console.log(section.campus);
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:139]: Here too. We should add some linter rules

File: src/pages/courseListingEntries/index.ts [hunk 8]
```diff
@@ -39,95 +41,130 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
...
-      }
-      else{
-        section.instructionalMethod = 'in-person';
+      if (instructionalMethodRegex.test(sectionInfo)) {
+        section.instructionalMethod = instructionalMethodRegex
+          .exec(sectionInfo)![1]
+          .toLowerCase()
+          .trim();
       }
+      console.log(section.instructionalMethod);
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:153]: Here

File: src/pages/courseListingEntries/index.ts [hunk 9]
```diff
@@ -142,11 +179,10 @@ export const classScheduleListingExtractor = async ($: cheerio.Root): Promise<Se
         });
       }
       section.schedule = scheduleData;
-
       sections.push(section);
     }
     return sections;
   } catch (error) {
     throw new Error(`Failed to get sections: ${error}`);
   }
```
Comment (aomi) [inline src/pages/courseListingEntries/index.ts:187]: I don't believe there is any value in catching the entire function implementation. I think we should remove it.

**Review by aomi — APPROVED** [review]
Minor things but generally a nice improvement 💯 and it passes those tests!

**Pull request merged**